Better separate the smart proxy installation instructions between Katello and Foreman

ianballou opened the pull request:
This PR adds some better separation between vanilla (no Katello), content, and Satellite smart proxies.

I've removed the parts that say "Katello only" and hid those sections instead.

Let me know if I missed anything.

ehelms commented:
I'd suggest we drop `include::modules/proc_configuring-repositories-proxy.adoc[leveloffset=+1]` in favor of 
https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_configuring-repositories.adoc

ehelms commented on guides/common/modules/proc_installing-capsule-server-packages.adoc:
```diff
@@ -13,19 +13,27 @@ To install {SmartProxyServer}, complete the following steps:
 ----
 # {package-update}
 ----
-ifeval::["{build}" == "satellite"]
```
I'd suggest we either drop this file all together and use https://github.com/theforeman/foreman-documentation/blob/master/guides/common/modules/proc_installing-the-satellite-server-packages.adoc or create a similarly structure file for smart-proxy/capsule. As there is a lot of nuance missing here per OS.

ehelms commented:
The storage requirements will need an update to account for OS versions similar to https://docs.theforeman.org/nightly/Installing_Server_on_Red_Hat/index-katello.html#storage-requirements_foreman

ianballou commented:
@ehelms I updated the smart proxy storage docs.  Let me know how the numbers and columns look. I removed `/opt` from CentOS 8 to match them main install storage requirements, but there might be something I missed to add.

ehelms commented:
@ianballou Looks like you need a rebase.

lzap commented:
LGTM, this was rebased. Any objections to merge?

ehelms commented:
When rendered, this has a section titled `2.1. Registering to Foreman server` which talks about sub-man registering the host to Foreman before installing. That is not actually required and I wonder if it should be an optional step later in the process.

ehelms commented on guides/common/modules/proc_installing-capsule-server-packages.adoc:
```diff
@@ -28,4 +34,28 @@ ifndef::satellite[]
 ----
 # {package-install} foreman-proxy-content
 ----
+=======
```
I think this line is breaking rendering.

ianballou replied:
Looks like it, I've pushed the fix.

ianballou commented:
> When rendered, this has a section titled `2.1. Registering to Foreman server` which talks about sub-man registering the host to Foreman before installing. That is not actually required and I wonder if it should be an optional step later in the process.

@ehelms I agree that it's optional.  I feel like it's there in case the smart proxy-related packages are expected to be downloaded from synced repositories on the Katello server.  I feel like it makes more sense to have this first then to mention it later. Otherwise I'd argue to remove it completely.  What do you think about just adding a note mentioning that this step is optional?

ehelms commented:
> @ehelms I agree that it's optional. I feel like it's there in case the smart proxy-related packages are expected to be downloaded from synced repositories on the Katello server. I feel like it makes more sense to have this first then to mention it later. Otherwise I'd argue to remove it completely. What do you think about just adding a note mentioning that this step is optional?

I like either optional, with a note about if you plan to sync the Foreman and Katello repositories locally then you need to do this step or make this an advanced method that comes after the "default" way of using the hosted repositories directly.

ianballou commented:
Alright, so I fixed it by creating a separate `common/modules/proc_registering-capsule-to-satellite-server.adoc` file since I needed to get the "optional" warning underneath the bold title.

lzap commented:
@ehelms or @ekohl can we get the final blessing?

lzap commented:
I am about to branch off 2.4 series, what is the status please? I can hold off the branching process if this is something that should be in 2.4/4.0. Let me know, thanks.

ianballou commented:
@lzap I don't think this PR is worth holding back branching.

ekohl commented on guides/common/modules/proc_installing-capsule-server-packages.adoc:
```diff
@@ -28,4 +37,27 @@ ifndef::satellite[]
+endif::[]
+
+ifdef::foreman-el,katello,satellite[]
+== [[rhel-7-package]]Red Hat Enterprise Linux 7
```
For non-Satellite builds, can we combine it into RHEL 7 / CentOS 7 or are they really different?

ekohl commented on guides/common/modules/proc_installing-capsule-server-packages.adoc:
```diff
@@ -28,4 +37,27 @@ ifndef::satellite[]
+endif::[]
+
+ifdef::foreman-el,katello[]
+== [[centos-8-package]]CentOS 8
```
Don't we support RHEL 8 upstream?

ianballou replied:
I'm going to assume yes unless anyone else says otherwise.

ekohl reviewed (commented):
I think overall this is ok and probably already an improvement. My biggest note (also inline) is that upstream I'd like to see `RHEL x / CentOS x` where the instructions are identical.

Also a note that CentOS 8 is discouraged by the CentOS project. They now prefer explicit CentOS Linux 8 or CentOS Stream 8. There may be many more of these, but I want to raise awareness to it.

ianballou commented on guides/common/modules/proc_registering-capsule-to-satellite-server.adoc:
```diff
@@ -0,0 +1,53 @@
+
+.Proxy and Network Prerequisites
+* The {ProjectServer} base operating system must be able to resolve the host name of the {SmartProxy} base operating system and vice versa.
+ifndef::foreman-deb[]
```
I'll remove it.

ianballou commented on guides/common/modules/ref_capsule-storage-requirements.adoc:
```diff
@@ -7,6 +7,34 @@ These values are based on expected use case scenarios and can vary according to
+|====
+|Directory |Installation Size |Runtime Size
+|/var/lib/pulp/ |1 MB |300 GB
+|/var/opt/rh/rh-postgresql12/lib/pgsql |100 MB |10 GB
```
I'm not seeing what would need the variable. Let me know if my recent push helped at all with this.

jturel reviewed (approved):
LGTM

@ehelms @ekohl it appears that all feedback has been addressed for some time. is it OK to be merged?

melcorr commented:
Would you mind rebasing, and I will merge! TY

ianballou commented:
@melcorr I've rebased just now.
